addons/stock/models/stock_move_line.py

```diff
@@ -5,6 +5,7 @@
 
 from odoo import api, fields, models, tools, _
 from odoo.exceptions import UserError, ValidationError
+from odoo.tools import OrderedSet
```

Comment:
Well, it need to be deterministe/reproductible (`set` isn't) and also the order can be important in some situation (here I don't think so, but it always good to be careful to not change the way it works: `OrderedSet` usage == recordset union (but better complexity)). 
In the performance point of view the set is faster than the OrderedSet (because low-level data-structure + not save the order), but it has the same complexity (and I never see a bottleneck because of OrderedSet).